packages/apollo-engine-reporting/src/plugin.ts

```diff
@@ -138,7 +155,18 @@ export const plugin = <TContext>(
       let didResolveSource: boolean = false;
 
       return {
+        didResolveOperation(requestContext) {
+          if (typeof options.traceReporting === 'function') {
+            options.traceReporting(requestContext).then(shouldReportTrace => {
+              if (!shouldReportTrace) didEnd(requestContext, false);
+            });
+          }
+        },
```

Comment:
A suggestion which is a combination of both of my comments here would be:

```suggestion
        async didResolveOperation(requestContext) {
          if (typeof options.traceReporting !== "function") return;
          const shouldReportTrace = await options.traceReporting(requestContext);
          
          // Help the user understand they've returned an unexpected value,
          // which might be a subtle mistake.
          if (typeof shouldReportTrace !== "boolean") {
            (requestContext.logger || logger).warn(
              "The 'traceReporting' predicate function must return a boolean value.");
            return;
          }
          
          if (!shouldReportTrace) didEnd(requestContext, false);
        },
```